src/modules/commander/Commander.cpp

```diff
@@ -4393,3 +4340,71 @@ void Commander::estimator_check(bool *status_changed)
 	check_valid(lpos.timestamp, _param_com_pos_fs_delay.get() * 1_s, lpos.z_valid,
 		    &(status_flags.condition_local_altitude_valid), status_changed);
 }
+
+void
+Commander::offboard_control_update(bool &status_changed)
+{
+	const offboard_control_mode_s &offboard_control_mode = _offboard_control_mode_sub.get();
+
+	// if this is the first time entering OFFBOARD the subscription may not be active yet
+	bool force_update = false;
+
+	if (commander_state_s::MAIN_STATE_OFFBOARD) {
+		if (offboard_control_mode.timestamp == 0) {
+			_offboard_control_mode_sub.forceInit();
+			force_update = true;
+		}
+	}
+
+	if (_offboard_control_mode_sub.updated() || force_update) {
```

Comment:
So `updated()` doesn't work as you would expect it to intuitively and you need to do a `forceInit()`?

Reply:
Explained below (hopefully). https://github.com/PX4/Firmware/pull/12242#issuecomment-501294935